porcupine/plugins/filetypes.py

```diff
@@ -252,6 +253,30 @@ def parse_filetype_name(name: str) -> FileType:
     )
 
 
+filetypes_var: tkinter.StringVar
+
+
+def _sync_filetypes_menu(event: object) -> None:
+    tab = get_tab_manager().select()
+    if isinstance(tab, tabs.FileTab):
+        try:
+            filetype_name = tab.settings.get("filetype_name", str)
+            if filetypes_var and filetype_name:
+                filetypes_var.set(filetype_name)
+        except KeyError:
```

Comment:
This has a weird bug. The default filetype for new files is Python, but if I make a new file, the filetype doesn't show up as Python until I switch to some other tab and back.

I think what's happening is:
1. New tab is created
2. Tab changes, `filetypes_var` is set to empty string because `filetype_name` isn't set yet
3. Filetype is set, but we no longer sync `filetypes_var` until changing tab.

I haven't yet checked whether this is correct. If you need help with debugging, I will help :)

One possible solution is to run the sync whenever the `filetype_name` setting changes. We can do it by binding `<<TabSettingChanged:filetype_name>>` inside `on_new_filetab()`. The documentation of `<<TabSettingChanged:...>>` events (generated from docstrings) is here: https://akuli.github.io/porcupine/tabs.html#porcupine.tabs.FileTab.TabSettingChanged-foo

Reply:
This seems to affect more than just new files. Try clicking the "Edit filetypes.toml" button, for example: it opens the file, but the filetypes menu doesn't show TOML as selected until you change tabs.